fix: proxy more parts of fetch Response

Techn1x opened the pull request:
resolves https://github.com/emberjs/ember-test-waiters/issues/518

Techn1x committed:
fix: proxy more parts of fetch Response

NullVoxPopuli commented:
Sorry it took me so long to get to this -- I forgot :( 

Can you address the type issue? thanks!!

Techn1x committed:
refactor: improve types in wait-for-fetch

Techn1x committed:
refactor: improve types again, add body prop

Techn1x commented:
I've gone a bit further than asked and;
* fixed the type issue, and ensured that the property names we're using are indeed properties of Response class, by using type guards
* added some code comments. I tend to over-comment so please let me know if I should reduce them.
* added response `body` property to list of properties

Probably a little more challenging to review now, sorry!

@NullVoxPopuli

Techn1x commented:
FYI @mfeckie - we can remove some ember/test-waiters pnpm patches we have with this update in the 4.1.1 release. Shoot me a message if it causes trouble
